src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/config/OAuth2ResourceServerSecurityConfiguration.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.core.convert.converter.Converter
+import org.springframework.http.HttpMethod
+import org.springframework.security.authentication.AbstractAuthenticationToken
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
+import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter
+import org.springframework.security.web.SecurityFilterChain
+
+@EnableWebSecurity
+class OAuth2ResourceServerSecurityConfiguration {
```

Comment:
We have tended to reverse this, only allow authenticated requests with but allow only certain requests through unauthenticated (e.g. health)

https://github.com/ministryofjustice/hmpps-nomis-prisoner-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/config/ResourceServerConfiguration.kt

Reply:
@mikehalmamoj had a similar comment, interested in your thoughts on my response also: https://github.com/ministryofjustice/approved-premises-api/pull/23#discussion_r940097738